frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/EmbedModalContent.tsx

```diff
@@ -0,0 +1,234 @@
+import { useState } from "react";
+import _ from "underscore";
+import { getSignedPreviewUrl, getSignedToken } from "metabase/public/lib/embed";
+import { getSetting } from "metabase/selectors/settings";
+import { useSelector } from "metabase/lib/redux";
+import type { ExportFormatType } from "metabase/dashboard/components/PublicLinkPopover/types";
+
+import { StaticEmbedSetupPane } from "../StaticEmbedSetupPane";
+import type {
+  ActivePreviewPane,
+  EmbeddingDisplayOptions,
+  EmbeddingParameters,
+  EmbeddingParametersValues,
+  EmbedResource,
+  EmbedResourceParameter,
+  EmbedResourceType,
+  EmbedType,
+} from "../EmbedModal.types";
+import { SelectEmbedTypePane } from "../SelectEmbedTypePane";
+import { EmbedModalContentStatusBar } from "./EmbedModalContentStatusBar";
+
+const FALLBACK_SECRET_KEY = "<METABASE_SECRET_KEY>";
+
+export interface EmbedModalContentProps {
+  embedType: EmbedType;
+  setEmbedType: (type: EmbedType) => void;
+
+  resource: EmbedResource;
+  resourceType: EmbedResourceType;
+  resourceParameters: EmbedResourceParameter[];
+
+  onUpdateEnableEmbedding: (enableEmbedding: boolean) => void;
+  onUpdateEmbeddingParams: (embeddingParams: EmbeddingParameters) => void;
+
+  onCreatePublicLink: () => void;
+  onDeletePublicLink: () => void;
+  getPublicUrl: (
+    resource: EmbedResource,
+    extension?: ExportFormatType,
+  ) => string | null;
+
+  className?: string;
+}
+
+export const EmbedModalContent = (
+  props: EmbedModalContentProps,
+): JSX.Element => {
+  const {
+    embedType,
+    setEmbedType,
+    resource,
+    resourceType,
+    resourceParameters,
+    onUpdateEnableEmbedding,
+    onUpdateEmbeddingParams,
+    onCreatePublicLink,
+    onDeletePublicLink,
+    getPublicUrl,
+  } = props;
+
+  const [pane, setPane] = useState<ActivePreviewPane>("code");
+
+  const siteUrl = useSelector(state => getSetting(state, "site-url"));
+  const secretKey =
+    useSelector(state => getSetting(state, "embedding-secret-key")) ||
```

Comment:
Are what you showed in these 2 videos possible in this PR? Or are these just to demonstrate if something went wrong you'd rather have shown the fallback key instead? (I've tested that there's it wasn't possible even for a new instance since you need to enable the embedding first before you can open the embed modal.)

If it's the latter, I'd still think using `checkNotNull` is more explicit and that tells the state of the codebase more correctly. As the code comment has suggested, this should never happen anyway.

Is there any reason why you don't want to use `checkNotNull` more often?